@here/harp-texture-atlas-generator/src/ImageVectorCoders.ts

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Jimp from "jimp";
+import { ColorUtils } from "./ColorUtils";
+import { FileSystem, ImageFormat } from "./FileSystem";
+import { ImageBitmapEncoderConstructor } from "./ImageBitmapCoders";
+import { Color, ImageDecoder, ImageDecoderConstructor } from "./ImageFactory";
+// tslint:disable-next-line: no-duplicate-imports
+import { ImageEncoder, ImageEncoderConstructor } from "./ImageFactory";
+
+// No @types for this module
+// tslint:disable:no-var-requires
+const svg2png = require("svg2png");
+
+export class ImageVectorDecoderConstructor implements ImageDecoderConstructor {
+    readonly targetWidth: number;
+    readonly targetHeight: number;
+
+    constructor(desiredWidth: number, desiredHeight: number) {
+        this.targetWidth = desiredWidth;
+        this.targetHeight = desiredHeight;
+    }
+
+    load(filePath: string): Promise<ImageDecoder> {
+        return new Promise((resolve, reject) => {
```

Comment:
This whole function can be rewritten with `async` to stg like this.
```
async load(...)
  const fileBuffer = await FileSystem.readFile(...)
  const pngBuffer = await svg2Png(...)
  const bitmap = await Jimp.read(pngBuffer)
  return new ImageVectorDecoder(...)
}
```
Use `async/await` for complex flows like this.